packages/@textlint/ast-node-types/package.json

```diff
@@ -2,8 +2,8 @@
   "name": "@textlint/ast-node-types",
   "version": "3.0.1",
   "description": "textlint AST node type definition.",
-  "main": "./lib/index.js",
-  "types": "./lib/index.d.ts",
+  "main": "./lib/src/index.js",
+  "types": "./lib/src/index.d.ts",
```

Comment:
> Correct main and types properties in package.json to prevent
> module resolving issue at @textlint/kernel

Probably, This changes is caused by adding `test/TextLintASTNodeTypes-test.ts`
I've noticed that [@textlint/ast-node-types/tsconfig.json](https://github.com/textlint/textlint/blob/master/packages/%40textlint/ast-node-types/tsconfig.json "@textlint/ast-node-types/tsconfig.json") does not specified `includes` target.

When no have `includes` setting in tsconfig.json, `tsc` convert all `.ts` files.
In other word, current config is `includes: ["*"]`.

> If the "files" and "include" are both left unspecified, the compiler defaults to including all TypeScript (.ts, .d.ts and .tsx)
> https://www.typescriptlang.org/docs/handbook/tsconfig-json.html

Alternative solution:

1. Revert this lines
2. Add `includes` to [@textlint/ast-node-types/tsconfig.json](https://github.com/textlint/textlint/blob/master/packages/%40textlint/ast-node-types/tsconfig.json "@textlint/ast-node-types/tsconfig.json")

```js
{
  "extends": "../../tsconfig.base.json",
  "compilerOptions": {
    "outDir": "./lib/"
  },
  "include": [
    "./src/**/*"
  ],
  "exclude": [
    ".git",
    "node_modules"
  ]
}
```

I prefer latter solution.

@0x6b How do you think?

Reply:
@azu, thank you. I have a lot to learn! Since I left the `tsconfig.json` as is, I had to modify `main` and `types` properties, but felt uncomfortable a bit 😒 for the directory structure (hey, `src` just under `lib`...?) I will push another commits which implement your solution.